tests/test_qubit_device.py

```diff
@@ -316,6 +316,21 @@ def test_parameters_accessed_outside_execution_context(self, mock_qubit_device):
 class TestExtractStatistics:
     """Test the statistics method"""
 
+    def test_observables_deprecated(self, mock_qubit_device_extract_stats, monkeypatch):
+        """Test that using a list of observables as an argument is deprecated."""
+        with monkeypatch.context() as m:
+            dev = mock_qubit_device_extract_stats()
+            with pytest.warns(
+                UserWarning,
+                match="The ``observables`` argument in ``QubitDevice.statistics`` is deprecated. ",
+            ):
+                dev.statistics([])
```

Comment:
Can we also test passing in both the `circuit` and `observables` arguments?

Reply:
Done! I also moved the `observables` argument of the `statistics` method at the beginning, because calling `dev.statistics([], circuit=qscript)` raised an error saying that `circuit` argument was used twice.